tests/common/pkt_filter/filter_pkt_in_buffer.py

```diff
@@ -0,0 +1,232 @@
+        pkt_dict = copy.deepcopy(pkt_dict)
+
+        for field, value in self.ignore_fields:
+            if pkt_dict.get(field):
+                pkt_dict[field].pop(value)
+
+        return pkt_dict
+
+
+    def __find_pkt_in_buffer(self):
+        """
+        Find expected packet in buffer by using matched fields
+
+        Returns:
+            Received packet
+        """
+        time.sleep(0.1)
+        common_buffer = self.ptfadapter.dataplane.packet_queues
+        packet_buffer = common_buffer[(0, self.dst_port_number)][:]
+
+        for pkt in packet_buffer:
+            packet_dict = convert_pkt_to_dict(Ether(pkt[0]))
+
+            for field, value in self.match_fields:
+                try:
+                    if packet_dict[field][value] != self.pkt_dict[field][value]:
+                        break
+                except KeyError:
+                    break
+            else:
+                return Ether(pkt[0])
+
+        return None
+
+
+    def __diff_between_dict(self, rcv_pkt_dict, exp_pkt_dict, path=''):
+        """
+        Find the difference between received packet dictionary and expected packet dictionary
+
+        Args:
+            rcv_pkt_dict: Received packet dictionary
+            exp_pkt_dict: Expected packet dictionary
+            path: Path to values of fields
+
+        Returns:
+            List of packet fields
+        """
+        for field in rcv_pkt_dict:
+            if field not in exp_pkt_dict:
+                self.received_pkt_diff.append("{}{}={}".format(path and path + ' ', field, rcv_pkt_dict[field]))
+            elif isinstance(rcv_pkt_dict[field], dict):
+                if path == "":
+                    path = field
+                else:
+                    path = path + " " + field
+                self.__diff_between_dict(rcv_pkt_dict[field], exp_pkt_dict[field], path)
+                path = ""
+            elif rcv_pkt_dict[field] != exp_pkt_dict[field]:
+                self.received_pkt_diff.append("{}{}={}".format(path + ' ', field, rcv_pkt_dict[field]))
+
+        return self.received_pkt_diff
+
+
+    def _diff_between_pkt(self, received_pkt):
+        """
+        Get the difference between received packet and expected packet
+
+        Args:
+            received_pkt: Received packet
+
+        Returns:
+            Difference between received packet and expected packet
+        """
+        received_pkt = convert_pkt_to_dict(received_pkt)
+        received_pkt = self.__remove_ignore_fields(received_pkt)
+
+        masked_pkt = self.__remove_ignore_fields(self.pkt_dict)
+
+        return self.__diff_between_dict(received_pkt, masked_pkt)
+
+
+    def filter_pkt_in_buffer(self):
+        """
+        Filter expected packet in buffer
+
+        Returns:
+            Bool value or difference between received packet and expected packet
+        """
+        self.received_pkt = self.__find_pkt_in_buffer()
+
+        if self.received_pkt:
+            self.__ignore_fields()
```

Comment:
This will update `self.masked_exp_pkt` in place. Maybe it is better to call `self.__ignore_fields()` in `__init__`.

Reply:
Good idea. Thanks!

Reply:
Done. Thanks!